### PR TITLE
Adds AB-Weeks

### LINKED ISSUE #316: [Feature]: A/B Week support
### What feature do you want in the App?

it would be nice for the timetable to hide or make obvious what subjects you do not have that week specifically (maybe make transparent)

### PR DESCRIPTION
**Pull request by Vito0912**
This PR introduces AB-Weeks for the timetable.

This fixes #316.

Importantly, this adds a third `FloatingActionButton`, which is far from ideal. However, I believe maintaining the ability to switch between views is crucial.

I could not find a way to implement this in the header of the SFCalendar. Additionally, the icons for the buttons are not optimal. Contribution is appriciated. 
Currently, it is implemented to always alternate between A and B weeks. However, I am not sure if there can be two consecutive "A" weeks. This would not affect the current timetable but could impact future timetables. If that is the case and we cannot find a way to determine how many A weeks there are, I propose defaulting to a random week for future lessons.

The week is determined by the badge on the website, so it should not matter if it is labeled A/B-Week, 1/2-Week, etc.

### CODE REVIEW

**Comment by TheBest6337** [comment]
It works well for me and is definitely better than my approach. However, the symbols could be improved, as I don’t think they clearly represent what the button does. Overall, it’s pretty good.

**Comment by Vito0912** [comment]
The icons definitely can and should be improved (I just can’t find the right icons xD). Actually, there shouldn’t be a third Floating Button at all. I think, by design, it should just be one, but I didn’t find a good place to integrate the thrid. 
But thanks for your feedback :)
Do you have icons in mind?

**Comment by TheBest6337** [comment]
I’ve also encountered a bug (not sure if it’s related to this PR or Lanis) where the substitutions switch places. It’s not a major issue, but it’s a bit annoying.
Normal:
image 
With AB Filter:
image

**Comment by Vito0912** [comment]
This should be ready now.

It displays the current week on the button at top right and allows switching between the current week and all weeks.
Design is the same as the button in `Calendar-View to have consistency.

Note: It is currently not possible to predict future weeks, so it defaults to the all weeks view.

**Comment by Kaanhehe** [comment]
I just tested it and it works almost.

My timetable has another badge not just a/b week but also "1.HJ". Simply it just classes I have in the first half of the year.
If I switch between A and B week the 1.HJ classes also get filtered out.

Also there is no B Week filter just A week, idk if its intended but seems weird.

File: app/lib/applets/timetable/student/student_timetable_view.dart [hunk 1]
```diff
@@ -134,12 +139,14 @@ class _StudentTimetableViewState extends State<StudentTimetableView> {
...
-                    logger.i(
-                        "Setting \"currentTimeTableView\" to \"${controller.view}\"");
-                    accountDatabase.kv
-                        .set("currentTimeTableView", "${controller.view}");
-                    _loadCalendarView();
+                  onViewChanged: (_) async {
+                    if (await _getCalendarView() != "${controller.view}") {
+                      logger.i(
+                          "Setting \"currentTimeTableView\" to \"${controller.view}\"");
+                      accountDatabase.kv
```
Comment (Vito0912) [inline app/lib/applets/timetable/student/student_timetable_view.dart:146]: Shouldn't the settings from the CombinedAppletBuilder be used, as it is specifically for this applet? (Like I did wrong in the prior PR)?

Comment (alessioC42) [reply app/lib/applets/timetable/student/student_timetable_view.dart:146]: Yes. This also looks like a setting that should be applet+user specific. The "accountDatabase" is shared between different user accounts

**Review by alessioC42 — APPROVED** [review]

**Pull request merged**